src/pages/Fox/components/FoxWifHat.tsx

```diff
@@ -1,25 +1,72 @@
-import { Box, Container, Heading, Image, useColorModeValue } from '@chakra-ui/react'
-import { foxAssetId } from '@shapeshiftoss/caip'
-import { useCallback } from 'react'
+import { Box, Container, Heading, Image, Skeleton, useColorModeValue } from '@chakra-ui/react'
+import type { AccountId } from '@shapeshiftoss/caip'
+import { foxWifHatAssetId, fromAccountId, fromAssetId } from '@shapeshiftoss/caip'
+import { bnOrZero } from '@shapeshiftoss/utils'
+import { useCallback, useMemo, useState } from 'react'
 import { useTranslate } from 'react-polyglot'
 import FoxWifHatIcon from 'assets/foxwifhat-logo.png'
 import { Text } from 'components/Text'
 import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { selectAccountIdsByChainId } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
 
+import { useFoxWifHatMerkleTreeQuery } from '../hooks/useFoxWifHatMerkleTreeQuery'
+import { FoxWifHatClaimModal } from './FoxWifHatClaimModal'
 import { FoxWifHatClaimRow } from './FoxWifHatClaimRow'
 
-// @TODO: replace with proper foxwifhat asset id
-const foxWifHatAssetId = foxAssetId
-
 export const FoxWifHat = () => {
   const translate = useTranslate()
   const isFoxWifHatEnabled = useFeatureFlag('FoxPageFoxWifHatSection')
   const containerBackground = useColorModeValue('blackAlpha.50', 'whiteAlpha.50')
+  const accountIdsByChainId = useAppSelector(selectAccountIdsByChainId)
+  const [isClaimModalOpened, setIsClaimModalOpened] = useState(false)
+  const [claimAccountId, setClaimAccountId] = useState<AccountId | undefined>()
+  const getFoxWifHatClaimsQuery = useFoxWifHatMerkleTreeQuery()
+
+  const handleClaimModalClose = useCallback(() => {
+    setClaimAccountId(undefined)
+    setIsClaimModalOpened(false)
+  }, [setClaimAccountId, setIsClaimModalOpened])
+
+  const handleClaimModalOpen = useCallback(
+    (accountId: AccountId) => {
+      setClaimAccountId(accountId)
+      setIsClaimModalOpened(true)
+    },
+    [setClaimAccountId, setIsClaimModalOpened],
+  )
+
+  const claimRows = useMemo(() => {
+    const accountIds = accountIdsByChainId[fromAssetId(foxWifHatAssetId).chainId]
 
-  const handleClaim = useCallback(() => {
-    // eslint-disable-next-line no-console
-    console.log('Claim')
-  }, [])
+    if (getFoxWifHatClaimsQuery.isLoading || !getFoxWifHatClaimsQuery.data)
```

Comment:
suggestion: `isFetching`

Reply:
Also, do we want to add an empty state for the `!data` case and decouple the right-side from here?
i.e if we've successfully fetched but we have no data, it means there's no claims for the current accounts set:

<img width="1466" alt="image" src="https://github.com/user-attachments/assets/880350ac-d8e3-497b-83aa-78a097aad647" />

Reply:
Nice catch, done in [7e9a198](https://github.com/shapeshift/web/pull/8611/commits/7e9a198a552c9ffbe24675fdb83a91d1875e342b)

Even though isLoading doesn't change anything in that particular case as the cache time is infinity